lib/tapioca/dsl/compiler.rb

```diff
@@ -74,11 +74,10 @@ def descendants_of(klass)
         sig { returns(T::Enumerable[T::Class[T.anything]]) }
         def all_classes
           @all_classes ||= T.let(
-            if @@requested_constants.any?
-              @@requested_constants.grep(Class)
-            else
-              ObjectSpace.each_object(Class)
-            end,
+            T.cast(
+              all_modules.select { |k| k.is_a?(Class) },
```

Comment:
I know it's kind of niche, but `.grep(Class)` is noticeably faster (last I benchmarked it), since it calls `Class#===` directly, without needing the overhead of a block call like `select` needs

```suggestion
              all_modules.grep(Class),
```

Reply:
I actually didn't know this! Good to know.

Reply:
2 other nice benefits:

1. It removes the need for [this cast](https://github.com/Shopify/tapioca/compare/85751a539fc2df50baaa258fe32a289f7d5213f2..2dfbfa43f5c0b984cb7bc7fdf229608f78868310) because `Enumerable#grep` has [a special overload for metatypes](https://github.com/sorbet/sorbet/blob/c1678be7c00cd2beb40adc153ccce5c424fd57a2/rbi/core/enumerable.rbi#L751-L757).
2. The `k.is_a?(Class)` was calling `is_a?` on `k`, which could be arbitrary user code that breaks things. `Class === k` would always be on a receiver we control, `Class`.

Reply:
Yes, @amomchilov is spot on, especially around the use of `===`. In this project, we prefer to use `===` over `is_a?` whenever it is possible, since the receiver of `===` is usually under our control, so the result can be more dependable.